### PR TITLE
Lineage scan

### PR DESCRIPTION
**Pull request by jar398**

### CODE REVIEW

File: ottreeindex/scripts/README.md [hunk 1]
```diff
@@ -3,18 +3,73 @@ This directory contains scripts for setting up the database from the command lin
 # Files:
 
 For command-line setup (for testing not from within pyramid):
+
 * config.yml: YAML file with database settings, including table names.
 * setup_db.py: creates DB tables if do not already exist; optionally deleting them all first
 * load_nexson.py: loads the nexson files into the database and creates the JSON index
-* create_otu_table.py: generates a csv file of otu-tree relationships and then loads this using the COPY command
+* create\_otu\_table.py: generates a csv file of otu-tree relationships and then loads this using the COPY command
+* initializedb.py: the pyramid file for initializing the database
```
Comment (kcranston) [inline ottreeindex/scripts/README.md]: initializedb.py should not be in this list, i.e. under the heading "command line setup (for testing not from within pyramid)"

File: ottreeindex/scripts/README.md [hunk 2]
```diff
@@ -3,18 +3,73 @@ This directory contains scripts for setting up the database from the command lin
...
+   (Do this only if the directory does not exist or is empty.)
+
+   `$ initdb /usr/local/var/postgres`
+
+1. Launch postgres DBMS server in background.
+
+    `$ launchctl load ~/Library/LaunchAgents/homebrew.mxcl.postgresql.plist`
 
-Pyramid setup:
-* initializedb.py: the pyramid file for iniatializing the database
```
Comment (kcranston) [inline ottreeindex/scripts/README.md]: should stay separated from other scripts, but should note that this gets run automatically by pyramid

File: ottreeindex/scripts/README.md [hunk 3]
```diff
@@ -3,18 +3,73 @@ This directory contains scripts for setting up the database from the command lin
...
+
+    `$ cp config.yml.example config.yml`
+
+Then adjust the new file for your local settings:
+
+* If Open Tree uses a database other than 'newoti', set this in config.yml.
+* If Open Tree runs as a user other than 'pguser', set this in config.yml.
+* Set location of taxonomy (usually a directory called 'ott' from an OTT distribution;
+  the directory containing taxonomy.tsv and so on).  The setup code
+  can be run without a taxonomy; just put None.
```
Comment (kcranston) [inline ottreeindex/scripts/README.md]: nice - thanks for thinking of that!

File: ottreeindex/scripts/create_otu_table.py [hunk 4]
```diff
@@ -43,7 +46,81 @@ def import_csv_file(connection,cursor,table,filename):
...
-def print_otu_file(connection,cursor,phy,nstudies=None):
+# All mapped OTUs in all studies
+def prepare_csv_files(connection,cursor,phy,taxonomy,nstudies=None):
+    otu_filename = "otu_mapping.csv"
+    tree_otu_filename = "tree_otu_assoc.csv"
+    seen_otus = {}
+    with open(otu_filename, 'w') as f, open(tree_otu_filename, 'w') as g:
+        fwriter = csv.writer(f)
+        gwriter = csv.writer(g)
+        # datafile format is 'ottid'\t'treeid' where treeid is not
```
Comment (jar398) [inline ottreeindex/scripts/create_otu_table.py]: I think that's comma, not tab

**Pull request merged**